pkg/api/socket.go

```diff
@@ -32,30 +30,15 @@ var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "api")
 
 // GetGroupIDByName returns the group ID for the given grpName.
 func GetGroupIDByName(grpName string) (int, error) {
```

Comment:
While at it, I think this func can be unexported since its only user is in the same package (`api.SetDefaultPermissions`)

Reply:
just curious, based on https://github.com/golang/go/blob/78c20c81aaa7d8d8a4aa2132806f51fb68f33a1e/src/os/user/cgo_lookup_unix.go#L6, there is another tag for cgo, as we disabled cgo in our build, do we need to add _osusergo_ tag ?

Reply:
@sayboras We e.g. build the `proxylib` with cgo but IMO we should make sure to still use the Go-only implementation even when building with cgo, so the behavior is consistent. Thus I think it's safer to always explicity specify `osusergo`.

Reply:
My understanding is that the osusergo tag effect is to use the go implementation when cgo is enabled, and has no effect when cgo is disabled.

_cgo_lookup_unix.go_ has `// +build cgo,!osusergo` meaning `cgo && !osusergo`  
_lookup_unix.go_ has `// +build !cgo osusergo` meaning `!cgo || osusergo`

Reply:
@kAworu @tklauser you are right. I overlooked this line https://github.com/golang/go/blob/master/src/os/user/lookup_unix.go#L6 --> it's OR condition :tada: